### PR TITLE
 Using Backend for Oidc Flow

### PR DESCRIPTION
**Pull request by mohityadav766**
### Describe your changes:

Adds [#15089](https://github.com/open-metadata/OpenMetadata/issues/15089)
- Implementation for Backend Auth Oidc Flow

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplication.java [hunk 1]
```diff
@@ -143,7 +150,7 @@ public void run(OpenMetadataApplicationConfig catalogConfig, Environment environ
     // Initialize HTTP and JDBI timers
     MicrometerBundleSingleton.initLatencyEvents(catalogConfig);
 
-    jdbi = createAndSetupJDBI(environment, catalogConfig.getDataSourceFactory());
+    this.jdbi = createAndSetupJDBI(environment, catalogConfig.getDataSourceFactory());
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplication.java:153]: can we remove "this", its not needed

File: openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+      String computedCallbackUrl = client.getCallbackUrl();
+      Map<String, List<String>> parameters = retrieveParameters(req);
+      AuthenticationResponse response =
+          AuthenticationResponseParser.parse(new URI(computedCallbackUrl), parameters);
+
+      if (response instanceof AuthenticationErrorResponse authenticationErrorResponse) {
+        LOG.error(
+            "Bad authentication response, error={}", authenticationErrorResponse.getErrorObject());
+        getErrorMessage(resp, new TechnicalException("Bad authentication response"));
+        return;
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java:73]: do we need to throw exception here, lets avoid having returns in the middle of the method

File: openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    if (preferredJwtClaim.contains("@")) {
+      userName = preferredJwtClaim.split("@")[0];
+    } else {
+      userName = preferredJwtClaim;
+    }
+
+    String url =
+        String.format(
+            "%s?id_token=%s&email=%s&name=%s",
+            "http://localhost:8585", credentials.getIdToken().getParsedString(), email, userName);
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java:192]: this I assume in progress?

File: openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplication.java (hunk 1 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplication.java]: done

File: openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java (hunk 3 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java]: yeah fixed this

File: openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java (hunk 2 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/security/AuthCallbackServlet.java]: removed

**Pull request merged**